### PR TITLE
Add templates

### PR DESCRIPTION
**Pull request by demenskyi**
Templates support added.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,22 +2,33 @@
...
-```
+```sh
 npx create-singlestoredb-app <YOUR_APP_NAME> <YOUR_API_KEY>
 ```
 
 4- Your app will be up and running on http://localhost:3000, powered by SingleStoreDB!
 
+### How to start using the template?
+
+1. Sign up for a free trial account at https://portal.singlestore.com
```
Comment (mivasconcelos) [inline README.md:19]: I don't think we need the signup step again

File: bin/start.js [hunk 2]
```diff
@@ -24,8 +26,18 @@ async function setupConnection(hostname, password) {
   }
 }
 
+if (options.template) {
+  if (!isValidTemplateName(options.template)) {
+    console.error("Invalid template name");
+    process.exit(1);
+  }
+
+  introMessage("Creating a SingleStore application");
```
Comment (mivasconcelos) [inline bin/start.js]: nit: Creating a SingleStore application with xxx template

**Comment by mivasconcelos** [comment]
```
~/s/create-singlestore-app ❯❯❯ npx create-singlestoredb-app demomarta 346237f528ddf9f3e1166c08ac4ff004c249919372bf2475b4eb826bafd74fe6 --template test
Invalid template name
~/s/create-singlestore-app ❯❯❯ npx create-singlestoredb-app demomarta 346237f528ddf9f3e1166c08ac4ff004c249919372bf2475b4eb826bafd74fe6 --template elegance-next

                           oo
                 ooOOOOOo     oOo
            ooOOOOOOOOOOOOOOo   oOOOo
         oOOOOOOOOOOOOOo oOOOOo  oOOOOo
       oOOOOOOOOOo            oOo  oOOOOo
      oOOOOOOOOo                 o  oOOOOo
     oOOOOOOOO                       oOOOOOo
    oOOOOOOOO                        oOOOOOo
    oOOOOOOOO                        oOOOOOo
    oOOOOOOOO                       oOOOOOOo
    oOOOOOOOOO                    oOOOOOOOOo
     oOOOOOOOOOo                 oOOOOOOOOo
      oOOOOOOOOOOo            oOOOOOOOOOOo
       oOOOOOOOOOOOOOOoo oOOOOOOOOOOOOOoo
         oOOOOOOOOOOOOOOOOOOOOOOOOOOOo
            ooOOOOOOOOOOOOOOOOOOOoo
                 ooooOOOOOOoooo

Creating a SingleStore application
    
✔ What is your project named? … my-app
✔ What connection type would like to use? › Kai

Cloning into 'my-app'...
```

It's looking good, nice job!

**Review by mivasconcelos — APPROVED** [review]

**Review by mivasconcelos — CHANGES_REQUESTED** [review]
Actually, it doesn't start the app or create the workspace like we do without the template, shouldn't it? 

```
~/s/create-singlestore-app ❯❯❯ npx create-singlestoredb-app demomarta 346237f528ddf9f3e1166c08ac4ff004c249919372bf2475b4eb826bafd74fe6 --template elegance-next

                           oo
                 ooOOOOOo     oOo
            ooOOOOOOOOOOOOOOo   oOOOo
         oOOOOOOOOOOOOOo oOOOOo  oOOOOo
       oOOOOOOOOOo            oOo  oOOOOo
      oOOOOOOOOo                 o  oOOOOo
     oOOOOOOOO                       oOOOOOo
    oOOOOOOOO                        oOOOOOo
    oOOOOOOOO                        oOOOOOo
    oOOOOOOOO                       oOOOOOOo
    oOOOOOOOOO                    oOOOOOOOOo
     oOOOOOOOOOo                 oOOOOOOOOo
      oOOOOOOOOOOo            oOOOOOOOOOOo
       oOOOOOOOOOOOOOOoo oOOOOOOOOOOOOOoo
         oOOOOOOOOOOOOOOOOOOOOOOOOOOOo
            ooOOOOOOOOOOOOOOOOOOOoo
                 ooooOOOOOOoooo

Creating a SingleStore application
    
✔ What is your project named? … my-app
✔ What connection type would like to use? › MySQL
Cloning into 'my-app'...

added 393 packages, and audited 394 packages in 55s

117 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
Your app is now ready!
```

**Comment by demenskyi** [comment]
> Actually, it doesn't start the app or create the workspace like we do without the template, shouldn't it?
> 
> ```
> ~/s/create-singlestore-app ❯❯❯ npx create-singlestoredb-app demomarta 346237f528ddf9f3e1166c08ac4ff004c249919372bf2475b4eb826bafd74fe6 --template elegance-next
> 
>                            oo
>                  ooOOOOOo     oOo
>             ooOOOOOOOOOOOOOOo   oOOOo
>          oOOOOOOOOOOOOOo oOOOOo  oOOOOo
>        oOOOOOOOOOo            oOo  oOOOOo
>       oOOOOOOOOo                 o  oOOOOo
>      oOOOOOOOO                       oOOOOOo
>     oOOOOOOOO                        oOOOOOo
>     oOOOOOOOO                        oOOOOOo
>     oOOOOOOOO                       oOOOOOOo
>     oOOOOOOOOO                    oOOOOOOOOo
>      oOOOOOOOOOo                 oOOOOOOOOo
>       oOOOOOOOOOOo            oOOOOOOOOOOo
>        oOOOOOOOOOOOOOOoo oOOOOOOOOOOOOOoo
>          oOOOOOOOOOOOOOOOOOOOOOOOOOOOo
>             ooOOOOOOOOOOOOOOOOOOOoo
>                  ooooOOOOOOoooo
> 
> Creating a SingleStore application
>     
> ✔ What is your project named? … my-app
> ✔ What connection type would like to use? › MySQL
> Cloning into 'my-app'...
> 
> added 393 packages, and audited 394 packages in 55s
> 
> 117 packages are looking for funding
>   run `npm fund` for details
> 
> found 0 vulnerabilities
> Your app is now ready!
> ```

Yes, that’s right. Because for the elegance people should use already existed workspaces and set env. variables manually. This is completely different flow.

**Comment by mivasconcelos** [comment]
> > Actually, it doesn't start the app or create the workspace like we do without the template, shouldn't it?
> > ```
> > ~/s/create-singlestore-app ❯❯❯ npx create-singlestoredb-app demomarta 346237f528ddf9f3e1166c08ac4ff004c249919372bf2475b4eb826bafd74fe6 --template elegance-next
> > 
> >                            oo
> >                  ooOOOOOo     oOo
> >             ooOOOOOOOOOOOOOOo   oOOOo
> >          oOOOOOOOOOOOOOo oOOOOo  oOOOOo
> >        oOOOOOOOOOo            oOo  oOOOOo
> >       oOOOOOOOOo                 o  oOOOOo
> >      oOOOOOOOO                       oOOOOOo
> >     oOOOOOOOO                        oOOOOOo
> >     oOOOOOOOO                        oOOOOOo
> >     oOOOOOOOO                       oOOOOOOo
> >     oOOOOOOOOO                    oOOOOOOOOo
> >      oOOOOOOOOOo                 oOOOOOOOOo
> >       oOOOOOOOOOOo            oOOOOOOOOOOo
> >        oOOOOOOOOOOOOOOoo oOOOOOOOOOOOOOoo
> >          oOOOOOOOOOOOOOOOOOOOOOOOOOOOo
> >             ooOOOOOOOOOOOOOOOOOOOoo
> >                  ooooOOOOOOoooo
> > 
> > Creating a SingleStore application
> >     
> > ✔ What is your project named? … my-app
> > ✔ What connection type would like to use? › MySQL
> > Cloning into 'my-app'...
> > 
> > added 393 packages, and audited 394 packages in 55s
> > 
> > 117 packages are looking for funding
> >   run `npm fund` for details
> > 
> > found 0 vulnerabilities
> > Your app is now ready!
> > ```
> 
> Yes, that’s right. Because for the elegance people should use already existed workspaces and set env. variables manually. This is completely different flow.

oh, ok - it was not clear to me.

**Review by mivasconcelos — DISMISSED** [review]

**Pull request merged**